internal/impl/qdrant/output.go

```diff
@@ -0,0 +1,243 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package qdrant
+
+import (
+	"context"
+	"fmt"
+
+	pb "github.com/qdrant/go-client/qdrant"
+	"github.com/redpanda-data/benthos/v4/public/bloblang"
+	"github.com/redpanda-data/benthos/v4/public/service"
+)
+
+const (
+	poFieldBatching       = "batching"
+	poFieldGrpcHost       = "grpc_host"
+	poFieldAPIToken       = "api_token"
+	poFieldUseTLS         = "tls"
+	poFieldCollectionName = "collection_name"
+	poFieldID             = "id"
+	poFieldVectorMapping  = "vector_mapping"
+	poFieldPayloadMapping = "payload_mapping"
+)
+
+func outputSpec() *service.ConfigSpec {
+	return service.NewConfigSpec().
+		Version("4.32.2").
+		Categories("AI").
+		Summary("Adds items to a [Qdrant](https://qdrant.tech/) collection").
+		Description(service.OutputPerformanceDocs(true, true)).
+		Fields(
+			service.NewOutputMaxInFlightField(),
+			service.NewBatchPolicyField(poFieldBatching),
+			service.NewStringField(poFieldGrpcHost).
+				Description("The gRPC host of the Qdrant collection. Defaults to 'localhost:6334'.").Default("localhost:6334"),
```

Comment:
Is this a good default? Maybe we should make it an example instead and make it required.

Reply:
That should be fine.

Reply:
This should be an example not a default, it's a small amount of work from the users perspective but having it explicit is good practice for any fields that represent a connection.

Reply:
Done in [`b25a024` (#2772)](https://github.com/redpanda-data/connect/pull/2772/commits/b25a0245b16588a51b481b1d3529d695190a2f73). Thank you.